packages/community/anthropic/src/provider.ts

```diff
@@ -0,0 +1,416 @@
+import Anthropic from '@anthropic-ai/sdk';
+import type {
+  MessageCreateParamsBase,
+  MessageCreateParams,
+  MessageCreateParamsNonStreaming,
+  MessageCreateParamsStreaming,
+  ToolChoice,
+  MessageParam,
+  RawMessageStartEvent,
+  RawContentBlockDeltaEvent,
+  RawContentBlockStartEvent,
+  RawMessageDeltaEvent,
+  ContentBlock,
+  TextBlock,
+  ToolUseBlock,
+  TextDelta,
+} from '@anthropic-ai/sdk/resources/messages';
+import { Stream } from '@anthropic-ai/sdk/streaming';
+import type { Logger } from '@nestjs/common';
+import type {
+  CompletionMetadata,
+  ICompletionProvider,
+  CompletionRequest,
+  CompletionResponse,
+  ResourceModelConfig,
+  AIConnection,
+  TokenMessage,
+  AIProviderConfig,
+} from '@vm-x-ai/completion-provider';
+import { BaseCompletionProvider, TokenCounter } from '@vm-x-ai/completion-provider';
+import { Span } from 'nestjs-otel';
+import type { Subject } from 'rxjs';
+
+const API_TIMEOUT_SECONDS = 300;
+
+export type CallAPIRequest = MessageCreateParamsNonStreaming | MessageCreateParamsStreaming;
+
+export type AnthropicConnectionConfig = {
+  apiKey: string;
+};
+
+export class AnthropicLLMProvider extends BaseCompletionProvider<Anthropic> implements ICompletionProvider {
+  constructor(logger: Logger, provider: AIProviderConfig) {
+    super(logger, provider);
+  }
+
+  @Span('Anthropic.getMaxReplyTokens')
+  getMaxReplyTokens(request: CompletionRequest): number {
+    const maxTok = request.config?.max_tokens || 100; // REPLACE THIS!!
```

Comment:
Should we replace the `100` value with something else?

Reply:
We should have a config setting in the ai provider that asks the user for a default max_tokens value. This should be mandatory for providers that require this field...